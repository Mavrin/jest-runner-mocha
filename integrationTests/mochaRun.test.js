const execa = require('execa');
const path = require('path');

const rootDir = path.join(__dirname, '..');

const runJest = fixture => {
  return execa(
    'jest',
    [
      '--useStderr',
      '--no-watchman',
      '--no-cache',
      '--projects',
      path.join(__dirname, '__fixtures__', fixture),
    ],
    {
      env: process.env,
    },
  ).catch(t => t);
};

const normalize = output =>
  output
    .replace(/\(?\d*\.?\d+m?s\)?/g, '')
    .replace(/, estimated/g, '')
    .replace(new RegExp(rootDir, 'g'), '/mocked-path-to-jest-runner-mocha')
    .replace(/\s+\n/g, '\n');

it('Works when it has only passing tests', async () => {
  const { stderr } = await runJest('passing');
  expect(normalize(stderr)).toMatchSnapshot();
});

it('Works when it has a failing tests', async () => {
  const { stderr } = await runJest('failing');
  expect(normalize(stderr)).toMatchSnapshot();
});

it('Works when it throws an error', async () => {
  const { stderr } = await runJest('errorInTest');
  expect(normalize(stderr)).toMatchSnapshot();
});

it('Works when it throws an error outside the tests', async () => {
  const { stderr } = await runJest('errorOusideTest');
  expect(normalize(stderr)).toMatchSnapshot();
});
