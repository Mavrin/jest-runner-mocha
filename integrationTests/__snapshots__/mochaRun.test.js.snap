// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Works when it has a failing tests 1`] = `
"FAIL integrationTests/__fixtures__/failing/__mocha__/__tests__/file.test.js
    My tests This test fails
      [31mAssertionError [ERR_ASSERTION]: 1 == 2[0m
      [32m+ expected[0m [31m- actual[0m
      [31m-1[0m
      [32m+2[0m
    at mocked-stack-trace
  ‚úï This test fails
Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:
Ran all test suites.
"
`;

exports[`Works when it has only passing tests 1`] = `
"PASS integrationTests/__fixtures__/passing/__mocha__/__tests__/file.test.js
  ‚úì This test passes
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:
Ran all test suites."
`;

exports[`Works when it throws an error 1`] = `
"FAIL integrationTests/__fixtures__/errorInTest/__mocha__/__tests__/file.test.js
  ‚óè Test suite failed to run
    Some error message
      at mocked-stack-trace
      at mocked-stack-trace
      at mocked-stack-trace
      at mocked-stack-trace
      at mocked-stack-trace
      at mocked-stack-trace
      at mocked-stack-trace
      at mocked-stack-trace
      at mocked-stack-trace
Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:
Ran all test suites.
"
`;

exports[`Works when it throws an error outside the tests 1`] = `
"Error: Can't find a root directory while resolving a config file path.
Provided path to resolve: /mocked-path-to-jest-runner-mocha/integrationTests/__fixtures__/errorOusideTest
cwd: /mocked-path-to-jest-runner-mocha
    at mocked-stack-trace
    at mocked-stack-trace
    at mocked-stack-trace
    at mocked-stack-trace
    at mocked-stack-trace
    at mocked-stack-trace
    at mocked-stack-trace
    at mocked-stack-trace
    at mocked-stack-trace
    at mocked-stack-trace
"
`;
